### PR TITLE
Use OAuth login

### PR DESCRIPTION
**Pull request by zachmullen**
This won't work until https://github.com/multinet-app/multinetjs/pull/28 is merged and a new release of multinetjs is cut.

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -12,10 +14,14 @@ Vue.config.productionTip = false;
...
 
-new Vue({
-  render: (h) => h(App),
-  router,
-  // Necessary to correctly use direct-vuex
-  store: store.original,
-  vuetify,
-}).$mount('#app');
+oauthClient.maybeRestoreLogin().then(() => {
+  Object.assign(api.client.axios.defaults.headers.common, oauthClient.authHeaders);
```
Comment (waxlamp) [inline src/main.ts:18]: @zachmullen, how did you know to access the axios instance's persistent request headers in this way? Is this "safe"?

Comment (zachmullen) [reply src/main.ts:18]: I believe I looked it up and this was the recommended approach. It seems to work -- I'm not sure under exactly what conditions it's safe or not.

**Comment by AlmightyYakob** [comment]
This PR is now dependent on a new release of `multinetjs` being cut. One thing to note is that while if a new release was cut, login would technically work, but not much else would, because the methods in `multinetjs` still need to be updated to access the new API. We'll need to do some coordinated development between `multinetjs`, the client and the API, in order to get things fully working. I'm thinking that while this is happening, we can instead cut pre-releases for any `girder4` stuff, making full releases once it makes sense to. We'd also need to follow a similar process with the viz apps. Thoughts @waxlamp?

**Comment by waxlamp** [comment]
> This PR is now dependent on a new release of `multinetjs` being cut. One thing to note is that while if a new release was cut, login would technically work, but not much else would, because the methods in `multinetjs` still need to be updated to access the new API. We'll need to do some coordinated development between `multinetjs`, the client and the API, in order to get things fully working. I'm thinking that while this is happening, we can instead cut pre-releases for any `girder4` stuff, making full releases once it makes sense to. We'd also need to follow a similar process with the viz apps. Thoughts @waxlamp?

This sounds like exactly the right approach. I will work on cutting a pre-release version of multinetjs today so that this PR works properly, and then we'll spin up Emi on how these things work so that she can begin to do some basic transformations of the client code and multinetjs to shift functionality over to Girder 4.

**Pull request merged**